test/test_sparse.py

```diff
@@ -1999,6 +2000,31 @@ def _test_sparse_mask_shape(self, nnz_x1, nnz_x2, shape_i, shape_v, dtype, devic
         self.assertEqual(self.safeToDense(y1), expected)
         self.assertEqual(self.safeToDense(y2), expected)
 
+    @dtypes(torch.double, torch.cdouble)
+    @skipIfCrossRef
+    def test_sparse_mask_backward(self, device, dtype):
+        from itertools import product, repeat
+
+        shape = (5, 5)
+        sparse_dims = len(shape)
+        nnzs = (0, 5, 15, 25)
+
+        lhs_data = torch.arange(25, device=device).reshape(shape).to(dtype).to_sparse(sparse_dims)
+        rhs_data = lhs_data.clone()
+
+        for nnz in nnzs:
+            for lhs_is_coalesced, rhs_is_coalesced in product(*repeat((True, False), 2)):
+                lhs = lhs_data.clone()
+                lhs._values()[nnz:].fill_(0)
+                lhs.requires_grad_(True)
+
+                rhs = rhs_data.clone()
+                rhs._values()[:-nnz].fill_(0)
```

Comment:
Hmm, so the gradcheck on `sparse_mask` is essentially tested with full tensors (`lhs._nnz() == rhs._nnz()` for all `nnz` values). Could you create `lhs` and `rhs` that are truly sparse, say, via
```python
lhs = torch.sparse_coo_tensor(lhs_data._indices()[:, :nnz], lhs_data._values()[:nnz], lhs_data.shape).clone().requires_grad_(True)
```
Similarly for `rhs`.